zul/src/archive/web/js/zul/mesh/Frozen.js

```diff
@@ -314,16 +314,19 @@ zul.mesh.Frozen = zk.$extends(zul.Widget, {
 		if (mesh.head) {
 			// set fixed size
 			var totalCols = mesh.head.nChildren,
-				hdrows = mesh.eheadrows.rows,
 				// B70-ZK-2071: Use mesh.head to get columns.
 				hdcells = mesh.head.$n().cells,
 				hdcol = mesh.ehdfaker.firstChild,
 				ftrows = mesh.foot ? mesh.efootrows : null,
-				ftcells = ftrows ? ftrows.rows[0].cells : null;
+				ftcells = ftrows ? ftrows.rows[0].cells : null,
+				cellsWidth = [];
+
+			for (var i = 0; i < totalCols; i++)
+				cellsWidth.push(jq.px0(hdcells[i].offsetWidth));
 
 			for (var faker, i = 0; hdcol && i < totalCols; hdcol = hdcol.nextSibling, i++) {
-				if (hdcol.style.width.indexOf('px') == -1) {
-					var sw = hdcol.style.width = jq.px0(hdcells[i].offsetWidth),
+				if (hdcol.style.width.indexOf('px') == -1 || this._syncFaker) {
```

Comment:
1. any timing to set `this._syncFaker` false?
2. or simplify the condition to always recalculate